Islandora-CLAW/islandora_demo
# Extracting frames from videos!

**GitHub Issue**: Resolves https://github.com/Islandora-CLAW/CLAW/issues/1005

Requires: https://github.com/Islandora-CLAW/islandora/pull/128

# What does this Pull Request do?

Adds an action to create an image from a Video using some neato `ffmpeg` options stuffed into the `X-Islandora-Args` header.  That image was added to the `Video Original File` context, so now thumbnails will be automatically extracted using the first frame from the video!

# How should this be tested?

- Make the same changes to your Homarus config as per https://github.com/Islandora-CLAW/Crayfish/pull/63.
- Update your islandora module with https://github.com/Islandora-CLAW/islandora/pull/128
- Pull in this PR
- Re-import the feature
- Make a repository item, give it a model of `Video`
- Add a video media to that item, give it a use of `Original File`.
- Wait a bit (you may also be generating a video derivative in parallel), and you should see a TN media for the repository item you created!

# Additional Notes:
Thanks to @Natkeeran, he figured this out.

# Interested parties
@Islandora-CLAW/committers
